source/extensions/filters/common/ext_authz/check_request_utils.h

```diff
@@ -140,7 +140,7 @@ class CheckRequestUtils {
       const Envoy::Http::RequestHeaderMap& headers, uint64_t max_request_bytes, bool pack_as_bytes,
       bool encode_raw_headers, const MatcherSharedPtr& request_header_matchers);
   static void setTLSSession(envoy::service::auth::v3::AttributeContext::TLSSession& session,
-                            const Ssl::ConnectionInfoConstSharedPtr ssl_info);
+                            OptRef<const Envoy::Network::Connection> connection);
```

Comment:
```suggestion
                            const Envoy::Network::Connection& connection);
```

It's not an optional parameter; it always has a value. I realize that `StreamFilterCallbacks::connection()` returns an OptRef, but it's already assumed to have a value and dereferenced in other places in `createHttpCheck()`, so just do the same for calling this function.

Reply:
https://github.com/envoyproxy/envoy/pull/34100/commits/a49951c222c53f3ef2cde3a1b9843582ca9b39d8